packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1241,6 +1251,64 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     }
   }
 
+  private _notifyPreviewChanged(prevState: IComboBoxState) {
+    const { onPendingValueChanged } = this.props;
+
+    if (!onPendingValueChanged) {
+      return;
+    }
+
+    const {
+      currentPendingValue,
+      currentOptions,
+      currentPendingValueValidIndex,
+      currentPendingValueValidIndexOnHover
+    } = this.state;
+
+    let sendUndefinedPendingValue = false;
+    let currentPendingIndex: number | undefined = undefined;
+    let pendingValue: string | undefined = undefined;
+
+    // Check first if hover index has changed
+    if (currentPendingValueValidIndexOnHover !== prevState.currentPendingValueValidIndexOnHover) {
+      if (this._indexWithinBounds(currentOptions, currentPendingValueValidIndexOnHover)) {
+        currentPendingIndex = currentPendingValueValidIndexOnHover;
+      } else {
+        sendUndefinedPendingValue = true;
```

Comment:
Consider changing this to true by default and then setting it to false inside of the if statements. It would remove the need for an else statement